File: src/main/java/com/skynight/scrcpy/ControlCenter.java
package com.skynight.scrcpy;

public class ControlCenter {

    private static ControlCenter instance;
    public static synchronized ControlCenter getInstance() {
        if (instance == null) {
            instance = new ControlCenter();
        }
        return instance;
    }

    private ControlCenter() {

    }

    public ControlListener controlListener;
    public synchronized void setControlListener(ControlListener controlListener) {
        this.controlListener = controlListener;
    }
    public ControlListener getControlListener() {
        return controlListener;
    }

    private volatile boolean wiredMethod = true;
    public synchronized void setWiredMethod(boolean wiredMethod) {
        this.wiredMethod = wiredMethod;
    }
    public boolean isWiredMethod() {
        return wiredMethod;
    }
}

File: src/main/kotlin/com/skynight/scrcpy/Base/ControlCenter.kt
package com.skynight.scrcpy.Base

import com.skynight.scrcpy.ControlListener


class ControlCenter private constructor() {

    @Volatile
    @set:Synchronized
    lateinit var controlListener: ControlListener

    @Volatile
    @set:Synchronized
    var isWiredMethod = true

    companion object {
        private var instance: ControlCenter? = null
        @Synchronized
        fun getInstance(): ControlCenter {
            if (instance == null) {
                instance = ControlCenter()
            }
            return instance as ControlCenter
        }
    }
}
